frontend/apps/cloudstorage/cloudstorage.lua

```diff
@@ -200,6 +216,18 @@ function CloudStorage:cloudFile(item, path)
                             text = _("Downloading may take several minutes…"),
                             timeout = 1,
                         })
+                    elseif self.type == "webdav" then
+                        local callback_close = function()
+                            self:onClose()
+                        end
+                        UIManager:scheduleIn(1, function()
+                            WebDav:downloadFile(item, self.address, self.username, self.password, path_dir, callback_close)
+                        end)
+                        UIManager:close(self.download_dialog)
+                        UIManager:show(InfoMessage:new{
+                            text = _("Downloading may take several minutes…"),
```

Comment:
I believe text is a member of the Button.

Reply:
I mean the string that's copied 10 times.

Reply:
Ah yes. Might take the opportunity to make it something nicer, "several minutes" seems a bit pessimistic for the type of files useful on an ebook reader.

Please wait while your file is downloading.
Download in progress.
?

Reply:
Sure, although at least for my ereader and file use it hardly seems pessimistic. (I tend to go for USB most of the time, and SSH otherwise.) My H2O doesn't come anywhere near the vicinity of the speed of my internet/wifi, regardless whether it's SD card speed/interface bound, wifi-chip bound or CPU bound. But my phone would indeed download a small 50 MB file in under 10 seconds.

I still think a little "don't panic" message might not be a bad idea though.

`Downloading. This might take a moment.`

Reply:
I've changed it.

The WiFi on the Clara is not amazing but too bad really. It didn't time exactly but a 5MB PDF took around 5s. I'm charging the old Kobo Mini to see if it much worse.